sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -329,22 +329,33 @@ public String getId() {
      * <p>
      * After subscription the operation will be performed. The {@link Flux} will
      * contain one or several feed response of the obtained items. In case of
-     * failure the {@link Flux} will error.
+     * failure the {@link CosmosContinuablePagedFlux} will error.
      *
      * @param <T> the type parameter
      * @param querySpec the SQL query specification.
      * @param options the feed options.
      * @param klass the class type
-     * @return a {@link Flux} containing one or several feed response pages of the obtained items or an error.
+     * @return a {@link CosmosContinuablePagedFlux} containing one or several feed response pages of the obtained items or an error.
      */
-    public <T> Flux<FeedResponse<T>> queryItems(SqlQuerySpec querySpec, FeedOptions options, Class<T> klass) {
-        return getDatabase().getDocClientWrapper().queryDocuments(getLink(),
-                                                                  querySpec, options)
-                   .map(response -> BridgeInternal.createFeedResponseWithQueryMetrics(
-                       (CosmosItemProperties
-                            .getTypedResultsFromV2Results((List<Document>) (Object) response.getResults(),
-                                                          klass)), response.getResponseHeaders(),
-                       response.queryMetrics()));
+    public <T> CosmosContinuablePagedFlux<T> queryItems(SqlQuerySpec querySpec, FeedOptions options, Class<T> klass) {
```

Comment:
aren't we going to have two options class types? now it is not clear if continuationToken is set in FeedOption and also passed by 'CosmosContinuablePageFlux.byPage(continuationToken)' who overrides who.

Reply:
I agree with the confusion as "who overrides who".
We don't need to necessarily split them. Since `continuationToken` is used internally from `feedOptions` class, which we don't need to change. 
I think a better solution here would be to hide public setter of `continuationToken` and `maxItemCount`. So that end users would have to go via `byPage(continuationToken, pageSize)` route to set them up, and internally we can copy them to feedOptions, and keep things as it is. We will not have to change the business logic for these.

Reply:
> some questions:
> 
> 1. Aren't we going to split the options as discussed earlier?
> 2. can you please run a perf benchmark with and without this PR using this benchmark, and see how different the numbers are?
>    #6771

Regarding options, let's discuss that offline. 
Regarding benchmark, I will run them today and will post the numbers here.